images/recycler/scripts/recycler.sh

```diff
@@ -22,8 +21,10 @@ if [[ ! -d ${dir} ]]; then
     exit 1
 fi
 
-# shred all files
-find ${dir} -type f -exec shred -fuvz {} \;
+find ${dir} -type f -exec stat -c "%u %g %n"  {} \; |xargs -n 3 bash -c 'sudo -u "#$1" shred  $3' argv0
```

Comment:
I wasn't sure... this script is expected to already running as uid 0, right?
Also, here is the test structures I threw at the script: https://gist.github.com/liggitt/02c67567bd01f8f9b0f5